### DISCUSSION TITLE
posts/2023/08/making-giscus-less-gabby/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#28] [Announcements]
# posts/2023/08/making-giscus-less-gabby/

The goal was to offer a comments section without forcing a massive JavaScript load on every visitor. Mission accomplished. Here’s how.

https://www.brycewray.com/posts/2023/08/making-giscus-less-gabby/?utm_campaign=RSS&utm_source=RSS&utm_medium=RSS

**Comment by bobmonsour** [comment]
Now. you have me thinking of entering the rathole of adding giscus comments to the 11tybundle.dev blog posts. Thanks for that...but nice work!

**Reply by brycewray** [reply]
@bobmonsour Thanks for the (semi-)kind words. 😄  The good thing is that there's only a tiny bit of Hugo-specific code in what I wrote — basically, switch out the `{{ $dataThemeLight }}` and `{{ $dataThemeDark }}` stuff — so, if you do want to use it with Eleventy, it should be easy to implement there.

**Edit, 2024-02-15**: I have since edited the post to separate the JS from Hugo templating, hoping to make it still easier for non-Hugo users.

**Reply by bobmonsour** [reply]
Thanks for the pointers. I think I'm going to save this for the fall, with summer travel and other things taking priority. I like the fact that the commenter has to be logged into GitHub to comment...keeping out spamming passersby. Thanks again. It really is nice work...seriously.

**Comment by Nooshu** [comment]
Thanks so much for this post, Bryce! I implemented the details / summary functionality yesterday, and it was only when testing my pages today I saw `client.js` still loading when opening the pages 🤬

I'll go through this post and hopefully replicate what you have done to fix it! 🤞

**Reply by Nooshu** [reply]
I implemented this today, works perfectly! Thanks so much for persevering and posting your solution! 

I've only changed one thing, which is to only inject the script tag once (e.g. if a user clicks on the show / hide button multiple times), not that I think it really matters, but Firefox likes to show a lot of scary looking console messages when the script is initialised. 

Code below if you are interested.

Added: `"class": "giscus-script"` to the `giscusAttributes` object then added an existence check in the script:

```js
// Is the script already in the DOM?
var scriptInDOM = document.getElementsByClassName('giscus-script');
// Check the scripts existence and only inject the script once into the DOM.
if(!scriptInDOM.length){
	// add the script to the page and load the comments
	divToAdd.appendChild(giscusScript);
} 
```
